packages/core/lib/console.js

```diff
@@ -130,6 +140,13 @@ class Console extends EventEmitter {
     abstractions.forEach(abstraction => {
       contextVars[abstraction.contract_name] = abstraction;
     });
+
+    this.context = this.setContextVars(contextVars);
+
+    // make sure the repl gets the new contracts in its context
+    if (this.repl) {
+      this.repl.context = this.context;
```

Comment:
I suspect this will overwrite any `repl.context` variables that `this.context` doesn't know about such as user defined variables as well as the account globals.

Reply:
Oops! That's what I get for trying to get in a quick fix. :) Thanks!

Reply:
So, it does. I set the other context variables that are needed after `provision()` is run.

Reply:
@cds-amal my last commit should address this. Let me know if you have any concerns!

Reply:
Oh I missed that!